### PR TITLE
Local storage

### LINKED ISSUE #34: LocalStorage
## Value Statement
As a developer, I want to save the products that the users like or the products they add to the cart.

## Acceptance Criteria
- [ ] When the page is refreshed, the products selected by the user should appear.

## Tasks
- [ ] Create feature branch
- [ ] Write functions

### PR DESCRIPTION
**Pull request by suademre**
I added LocalStorage. When the user adds a favorite Page or shopping card, their data is not lost when they refresh the page.

This resolved #34

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -14,6 +14,17 @@ function App() {
     const [favouriteItems, setFavouriteItems] = useState([]);
     const [basketItems, setBasketItems] = useState([]);
 
+    useEffect(() => {
+        let basketProducts = JSON.parse(
+            window.localStorage.getItem('basketProduct')
+        );
+        let favouriteProducts = JSON.parse(
+            window.localStorage.getItem('favouriteProduct')
+        );
```
Comment (mpagels) [inline src/App.js:23]: Why do you use `let` instead of `const`? You're not changing the values here.

File: src/App.js [hunk 2]
```diff
@@ -31,6 +42,10 @@ function App() {
             newFavorites = favouriteItems.concat(product);
         }
         setFavouriteItems(newFavorites);
+        window.localStorage.setItem(
+            'favouriteProduct',
+            JSON.stringify(newFavorites)
+        );
```
Comment (mpagels) [inline src/App.js:48]: You could write another `useEffect` that runs, when `favoriteItems` is changed.
Here is a [good article](https://blog.logrocket.com/using-localstorage-react-hooks/) about that topic.

File: src/App.js [hunk 3]
```diff
@@ -14,6 +14,17 @@ function App() {
     const [favouriteItems, setFavouriteItems] = useState([]);
     const [basketItems, setBasketItems] = useState([]);
 
+    useEffect(() => {
+        let basketProducts = JSON.parse(
+            window.localStorage.getItem('basketProduct')
```
Comment (mpagels) [inline src/App.js:19]: You don't need to access the `window` object. You can just call `localStorage.getItem`.

**Review by mpagels — APPROVED** [review]
Good Job Suad. I found some improvements you could implement, and I left a good article about implementing localStorage to a React project. Pls review.

File: src/App.js (hunk 1 above)
Comment (suademre) [reply src/App.js:23]: I changed

File: src/App.js (hunk 3 above)
Comment (suademre) [reply src/App.js:19]: I changed it

File: src/App.js (hunk 2 above)
Comment (suademre) [reply src/App.js]: thanx

**Pull request merged**